samtranslator/model/eventsources/push.py

```diff
@@ -530,13 +530,12 @@ def to_cloudformation(self, **kwargs):
     def _get_permissions(self, resources_to_link):
         permissions = []
 
-        permissions.append(self._get_permission(resources_to_link, "*", "Test"))
-
         # By default, implicit APIs get a stage called Prod. If the API event refers to an
         # explicit API using RestApiId property, we should grab the stage name of the explicit API
-        permitted_stage = suffix = "Prod"
+        # all stages for an API are given permission
+        permitted_stage = "*"
+        suffix = "Prod"
         if 'explicit_api_stage' in resources_to_link:
-            permitted_stage = resources_to_link['explicit_api_stage']['permitted_stage']
```

Comment:
If we're no longer using `permitted_stage` in `resources_to_link` anywhere, can you remove it from the `resources_to_link()` function above? Seems like that's just dead code now.

Reply:
removed the permitted_stage from the return dictionary of resources_to_link() function